### PR TITLE
Feature gate all caches

### PR DESCRIPTION
**Pull request by prestonvanloon**
As part of investigating https://github.com/prysmaticlabs/prysm/issues/3885, we want to disable all caches until we can carefully roll them out with detection of abnormalities in block production. 

- [x] Tested with local chain
- [x] Verified validator RPC calls are not too expensive with caches off.

### CODE REVIEW

File: beacon-chain/cache/committee.go [hunk 1]
```diff
@@ -67,6 +68,9 @@ func NewCommitteeCache() *CommitteeCache {
 // ShuffledIndices fetches the shuffled indices by epoch and shard. Every list of indices
 // represent one committee. Returns true if the list exists with epoch and shard. Otherwise returns false, nil.
 func (c *CommitteeCache) ShuffledIndices(epoch uint64, shard uint64) ([]uint64, error) {
+	if !featureconfig.Get().EnableShuffledIndexCache {
```
Comment (nisdas) [inline beacon-chain/cache/committee.go]: the committee cache is already gated by the new-cache flag if I am not wrong. 
@terencechain can you confirm ?

Comment (terencechain) [reply beacon-chain/cache/committee.go]: that's correct, we don't need this flag

Comment (prestonvanloon) [reply beacon-chain/cache/committee.go]: Then that flag needs to be moved here. Not putting the flag inside of the cache allows for misuse without proper flag control.

**Comment by terencechain** [comment]
Per discussion of last night, checkpoint state is KV store is required for fork choice as it's currently implemented in the fork choice spec. We should move `check_point.state` from `cache` package to `forkchoice` package

**Comment by prestonvanloon** [comment]
Confirmed locally that the chains advance normally with mixed nodes (cache on vs cache off) and that having the cache off is a minimal impact. I noticed only 1.4% difference in wall time for assignments with 128 validators in the system.

**Pull request merged**